src/store/reducers/movie-reducer.test.js

```diff
@@ -0,0 +1,90 @@
+import OperationStatus from '../../constants/operation-status';
+
+import ActionType from '../action-type';
+import movieReducer from './movie-reducer';
+
+describe(`Movie reducer`, () => {
+
+  test(`Reducer without additional parameters should return initial state`, () => {
+    expect(movieReducer(undefined, {}))
+      .toEqual({
+        moviesStatus: OperationStatus.UNSET,
+        movies: [],
+        promoStatus: OperationStatus.UNSET,
+        promoMovie: undefined,
+      });
+  });
+
+  test(`Reducer should update movies request status by setting appropriate operation status`, () => {
+    Object.values(OperationStatus).forEach((operationStatus) => {
+      const action = {
+        type: ActionType.SET_MOVIES_STATUS,
+        payload: operationStatus,
+      };
+      expect(movieReducer(undefined, action))
+        .toEqual({
+          moviesStatus: operationStatus,
+          movies: [],
+          promoStatus: OperationStatus.UNSET,
+          promoMovie: undefined,
+        });
```

Comment:
Чтобы каждый раз не дублировать, можно тестировать только то, что меняется.
```suggestion
    expect(movieReducer(undefined, action).moviesStatus).toEqual(operationStatus);
```

Reply:
1. Тогда видимо нужно будет проверять через временную переменную, так как, как правило, приходится проверять пару operationStatus+operationResult.
2. Так я дополнительно проверяю, что редьюсер не поменял ничего лишнего.

Reply:
Ок, но тоже выглядит как оверинжиниринг)